### ISSUE TITLE
Build failure with msvc 2017

### ISSUE THREAD

**Issue by alatiera** [harfbuzz/harfbuzz#2930]
Stumbled across this today in GStreamer's CI:

Its very recent, though not sure the exact commit that introduce it, will update the issue once I find out.

https://gitlab.freedesktop.org/cxw42/gstreamer/-/jobs/8512558

```
[3384/6184] Compiling C++ object subprojects/pango/tests/cxx-test.exe.p/cxx-test.cpp.obj
[3385/6184] Compiling C++ object subprojects/harfbuzz/src/harfbuzz.dll.p/hb-ot-layout.cc.obj
FAILED: subprojects/harfbuzz/src/harfbuzz.dll.p/hb-ot-layout.cc.obj 
"cl" "-Isubprojects\harfbuzz\src\harfbuzz.dll.p" "-Isubprojects\harfbuzz\src" "-I..\subprojects\harfbuzz\src" "-Isubprojects\harfbuzz" "-I..\subprojects\harfbuzz" "-I..\subprojects\freetype2\include" "-Isubprojects\zlib-1.2.11" "-I..\subprojects\zlib-1.2.11" "-Isubprojects\libpng-1.6.37" "-I..\subprojects\libpng-1.6.37" "-Isubprojects\glib" "-I..\subprojects\glib" "-Isubprojects\glib\glib" "-I..\subprojects\glib\glib" "-Isubprojects\proxy-libintl" "-I..\subprojects\proxy-libintl" "/MD" "/nologo" "/showIncludes" "/W2" "/EHsc" "/O2" "/Ob1" "/Zi" "/wd4018" "/wd4146" "/wd4244" "/wd4305" "/utf-8" "-DHAVE_CONFIG_H" "/utf-8" "-DHB_DLL_EXPORT" "/Fdsubprojects\harfbuzz\src\harfbuzz.dll.p\hb-ot-layout.cc.pdb" /Fosubprojects/harfbuzz/src/harfbuzz.dll.p/hb-ot-layout.cc.obj "/c" ../subprojects/harfbuzz/src/hb-ot-layout.cc
c:\builds\cxw42\gstreamer\gst-build\subprojects\harfbuzz\src\hb-ot-layout-common.hh(1427): error C2672: 'hb_set_digest_combiner_t<hb_set_digest_lowest_bits_t<unsigned long,4>,hb_set_digest_combiner_t<hb_set_digest_lowest_bits_t<unsigned long,0>,hb_set_digest_lowest_bits_t<unsigned long,9>>>::add_sorted_array': no matching overloaded function found
c:\builds\cxw42\gstreamer\gst-build\subprojects\harfbuzz\src\hb-ot-layout-common.hh(1729): note: see reference to function template instantiation 'bool OT::CoverageFormat1::collect_coverage<set_t>(set_t *) const' being compiled
        with
        [
            set_t=hb_set_digest_t
        ]
c:\builds\cxw42\gstreamer\gst-build\subprojects\harfbuzz\src\hb-ot-layout-gsubgpos.hh(374): note: see reference to function template instantiation 'bool OT::Coverage::collect_coverage<set_t>(set_t *) const' being compiled
        with
        [
            set_t=hb_set_digest_t
        ]
c:\builds\cxw42\gstreamer\gst-build\subprojects\harfbuzz\src\hb-ot-layout-gsubgpos.hh(373): note: while compiling class template member function 'bool OT::hb_collect_coverage_context_t<set_t>::stop_sublookup_iteration(OT::hb_collect_coverage_context_t<set_t>::return_t) const'
        with
        [
            set_t=hb_set_digest_t
        ]
c:\builds\cxw42\gstreamer\gst-build\subprojects\harfbuzz\src\hb-ot-layout-common.hh(1253): note: see reference to function template instantiation 'bool OT::hb_collect_coverage_context_t<set_t>::stop_sublookup_iteration(OT::hb_collect_coverage_context_t<set_t>::return_t) const' being compiled
        with
        [
            set_t=hb_set_digest_t
        ]
c:\builds\cxw42\gstreamer\gst-build\subprojects\harfbuzz\src\hb-ot-layout-gsub-table.hh(1543): note: see reference to class template instantiation 'OT::hb_collect_coverage_context_t<set_t>' being compiled
        with
        [
            set_t=hb_set_digest_t
        ]
c:\builds\cxw42\gstreamer\gst-build\subprojects\harfbuzz\src\hb-ot-layout-gsubgpos.hh(3493): note: see reference to function template instantiation 'void OT::SubstLookup::collect_coverage<hb_set_digest_t>(set_t *) const' being compiled
        with
        [
            set_t=hb_set_digest_t
        ]
c:\builds\cxw42\gstreamer\gst-build\subprojects\harfbuzz\src\hb-ot-layout-gsubgpos.hh(3798): note: see reference to function template instantiation 'void OT::hb_ot_layout_lookup_accelerator_t::init<OT::SubstLookup>(const TLookup &)' being compiled
        with
        [
            TLookup=OT::SubstLookup
        ]
c:\builds\cxw42\gstreamer\gst-build\subprojects\harfbuzz\src\hb-ot-layout-gsubgpos.hh(3779): note: while compiling class template member function 'void OT::GSUBGPOS::accelerator_t<OT::GSUB>::init(hb_face_t *)'
c:\builds\cxw42\gstreamer\gst-build\subprojects\harfbuzz\src\hb-machinery.hh(247): note: see reference to function template instantiation 'void OT::GSUBGPOS::accelerator_t<OT::GSUB>::init(hb_face_t *)' being compiled
c:\builds\cxw42\gstreamer\gst-build\subprojects\harfbuzz\src\hb-ot-layout-gsub-table.hh(1687): note: see reference to class template instantiation 'OT::GSUBGPOS::accelerator_t<OT::GSUB>' being compiled
c:\builds\cxw42\gstreamer\gst-build\subprojects\harfbuzz\src\hb-ot-layout-common.hh(1427): error C2780: 'bool hb_set_digest_combiner_t<hb_set_digest_lowest_bits_t<unsigned long,4>,hb_set_digest_combiner_t<hb_set_digest_lowest_bits_t<unsigned long,0>,hb_set_digest_lowest_bits_t<unsigned long,9>>>::add_sorted_array(const T *,unsigned int,unsigned int)': expects 3 arguments - 1 provided
c:\builds\cxw42\gstreamer\gst-build\subprojects\harfbuzz\src\hb-set-digest.hh(138): note: see declaration of 'hb_set_digest_combiner_t<hb_set_digest_lowest_bits_t<unsigned long,4>,hb_set_digest_combiner_t<hb_set_digest_lowest_bits_t<unsigned long,0>,hb_set_digest_lowest_bits_t<unsigned long,9>>>::add_sorted_array'
[3386/6184] Compiling C++ object subprojects/harfbuzz/src/harfbuzz.dll.p/hb-aat-layout.cc.obj
[3387/6184] Compiling C++ object subprojects/harfbuzz/src/harfbuzz.dll.p/hb-ot-face.cc.obj
[3388/6184] Compiling C++ object subprojects/harfbuzz/src/harfbuzz-subset.dll.p/hb-subset-plan.cc.obj
[3389/6184] Compiling C++ object subprojects/harfbuzz/src/harfbuzz-subset.dll.p/hb-subset.cc.obj
ninja: build stopped: subcommand failed.
section_end:1617343825:step_script
```

**Comment by alatiera** [comment]
I think this might be due to the same commit as #2928 092094f7055c1632e9bdf2c07f16e984a88a65ec

**Comment by behdad** [comment]
Oops. Sorry. Should be fixed.

**Comment by alatiera** [comment]
Indeed fixed by d3a2f999e4d5c83e14f6380b780aab170d9cbd39
